lib/api/decorators/property_schema_representer.rb

```diff
@@ -32,39 +32,31 @@
 
 module API
   module Decorators
-    class AllowedReferenceLinkRepresenter < Roar::Decorator
-      include Roar::JSON::HAL
+    class PropertySchemaRepresenter < ::API::Decorators::Single
+      def initialize(type: nil, name: nil, current_user: nil)
```

Comment:
@ulferts regarding your comment about the too large interface.
I did some things to improve the situation here:

* have a single base class containing the afforementioned properties (instead two classes + one helper)
* use named parameters to avoid confusion about the meaning of parameters
* kick the bools out of the initializer as they are optional, they can be assigned by the caller after initialization